Sources/Rownd/Models/PasskeyCoordinator.swift

```diff
@@ -0,0 +1,353 @@
+
+/*
+See LICENSE folder for this sample’s licensing information.
+
+Abstract:
+The authentication manager object.
+*/
+
+import AuthenticationServices
+import Foundation
+import os
+import Get
+import LocalAuthentication
+
+extension Data {
+    init?(base64EncodedURLSafe string: String, options: Base64DecodingOptions = []) {
+        let string = string
+            .replacingOccurrences(of: "-", with: "+")
+            .replacingOccurrences(of: "_", with: "/")
+            .padding(toLength: ((string.count+3)/4)*4,
+                  withPad: "=",
+                  startingAt: 0)
+
+        self.init(base64Encoded: string, options: options)
+    }
+
+    func base64URLEncodedString() -> String {
+      let string = self.base64EncodedString()
+      return string
+        .replacingOccurrences(of: "+", with: "-")
+        .replacingOccurrences(of: "/", with: "_")
+        .replacingOccurrences(of: "=", with: "")
+    }
+}
+
+extension LAContext {
+    enum BiometricType: String {
+        case none
+        case touchID
+        case faceID
+    }
+
+    var biometricType: BiometricType {
+        var error: NSError?
+
+        guard self.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: &error) else {
+            // Capture these recoverable error thru Crashlytics
+            return .none
+        }
+
+        if #available(iOS 11.0, *) {
+            switch self.biometryType {
+            case .none:
+                return .none
+            case .touchID:
+                return .touchID
+            case .faceID:
+                return .faceID
+            @unknown default:
+                return .none
+            }
+        } else {
+            return  self.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: nil) ? .touchID : .none
+        }
+    }
+}
+
+enum PasskeyCoordinatorMethods {
+    case SignIn
+    case SignUp
+}
+
+class PasskeyCoordinator: NSObject, ASAuthorizationControllerPresentationContextProviding, ASAuthorizationControllerDelegate {
+    
+    var method: PasskeyCoordinatorMethods? = nil
+    
+    private func getWindowScene() -> UIWindowScene? {
+        let scenes = UIApplication.shared.connectedScenes
+        return scenes.first as? UIWindowScene
+    }
+    
+    func signUpWith() {
+        //Add passkey to the Rownd user
+        method = PasskeyCoordinatorMethods.SignUp
+        let anchor: ASPresentationAnchor = (getWindowScene()?.windows.last?.rootViewController?.view.window)!
+        let bottomSheetController = getWindowScene()?.windows.first?.rootViewController?.presentedViewController as! BottomSheetController
```

Comment:
I wonder if we should just require this to be passed in as a function argument so we don't have to force typecast it, which could result in a crash.